GVFS/GVFS.Platform.Linux/LinuxFileSystemVirtualizer.cs

```diff
@@ -0,0 +1,719 @@
+                        if (gitCommand.IsValidGitCommand)
+                        {
+                            EventMetadata metadata = this.CreateEventMetadata(relativePath);
+                            metadata.Add(nameof(lockedGitCommand), lockedGitCommand);
+                            metadata.Add(TracingConstants.MessageKey.InfoMessage, "Git command created new folder");
+                            this.Context.Tracer.RelatedEvent(EventLevel.Informational, $"{nameof(this.OnNewFileCreated)}_GitCreatedFolder", metadata);
+
+                            // Record this folder as expanded so that GitIndexProjection will re-expand the folder
+                            // when the projection change completes.
+                            //
+                            // Git creates new folders when there are files that it needs to create.
+                            // However, git will only create files that are in ModifiedPaths.dat.  There could
+                            // be other files in the projection (that were not created by git) and so VFS must re-expand the
+                            // newly created folder to ensure that all files are written to disk.
+                            this.FileSystemCallbacks.OnPlaceholderFolderExpanded(relativePath);
+                        }
+                        else
+                        {
+                            this.FileSystemCallbacks.OnFolderCreated(relativePath);
+                        }
+                    }
+                    else
+                    {
+                        this.FileSystemCallbacks.OnFileCreated(relativePath);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                EventMetadata metadata = this.CreateEventMetadata(relativePath, e);
+                metadata.Add("isDirectory", isDirectory);
+                this.LogUnhandledExceptionAndExit(nameof(this.OnNewFileCreated), metadata);
+            }
+        }
+
+        private void OnFileRenamed(string relativeDestinationPath, bool isDirectory)
+        {
+            // TODO(Linux): VFSForGit doesn't need the source path on Linux for correct behavior,
+            // but it is available if required in the future
+            this.OnFileRenamed(
+                relativeSourcePath: string.Empty,
+                relativeDestinationPath: relativeDestinationPath,
+                isDirectory: isDirectory);
+        }
+
+        private void OnHardLinkCreated(string relativeNewLinkPath)
+        {
+            this.OnHardLinkCreated(
+                relativeExistingFilePath: string.Empty,
+                relativeNewLinkPath: relativeNewLinkPath);
+        }
+
+        private Result OnEnumerateDirectory(
+            ulong commandId,
+            string relativePath,
+            int triggeringProcessId,
+            string triggeringProcessName)
+        {
+            relativePath = ConvertDotPath(relativePath);
```

Comment:
Rather than making VFS4G call `ConvertDotPath` can your filesystem layer perform this conversion before calling `OnEnumerateDirectory`?

Reply:
👍